src/hotspot/share/gc/shared/locationPrinter.inline.hpp

```diff
@@ -51,8 +51,10 @@ oop BlockLocationPrinter<CollectedHeapT>::base_oop_or_null(void* addr) {
 
 template <typename CollectedHeapT>
 bool BlockLocationPrinter<CollectedHeapT>::print_location(outputStream* st, void* addr) {
+  bool in_heap = false;
   // Check if addr points into Java heap.
   if (CollectedHeapT::heap()->is_in(addr)) {
+    // base_oop_or_null() might be unimplemented and return NULL for some GCs/generations
```

Comment:
In such cases where the flag that we later set is dependent on the complete condition, it seems nicer to assign the result of the condition to it right away. That saves the assignment later too, having only a single assignment to it. Ymmv.
```suggestion
  // Check if addr points into Java heap.
  bool in_heap = CollectedHeapT::heap()->is_in(addr);
  if (in_heap) {
    // base_oop_or_null() might be unimplemented and return NULL for some GCs/generations.
```

(And drop the assignment to `in_heap` later).

Reply:
Thanks for looking at this PR. Your suggestion sound like a reasonable simplification. I've updated the code accordingly.